xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/XWikiAction.java

```diff
@@ -1221,4 +1158,27 @@ protected boolean readFromTemplate(XWikiDocument document, String template, XWik
 
         return false;
     }
+
+    /**
+     * Loop over the {@link RedirectionFilter} components until one of them perform a redirection. If none of the does,
+     * the action continues normally.
+     *
+     * @param context the current wiki content
+     * @return {@code true} if a redirection has been performed, {@code false} otherwise
+     * @throws XWikiException in case of error during the execution of a redirection filter
+     */
+    private boolean handleRedirect(XWikiContext context) throws XWikiException
+    {
+        // If no redirection are expected, this step is skipped.
+        if (this.handleRedirectObject) {
```

Comment:
This variable should maybe be deprecated for something a little more generic.

Reply:
Should it still be used by `RedirectClassRedirectionFilter`, or simply renamed into a more generic `redirectionsAllowed` field for instance?

Reply:
Note that you have to keep this variable since it's an API, my suggestion was more to deprecate it and introduce one more (or a different system to indicate this information like a method to overwite depending on what you prefer) that would have a more accurate naming.

Reply:
I've introduced an `supportRedirections` method instead of `handleRedirectObject` (moved to legacy)